lib/util.js

```diff
@@ -424,21 +438,51 @@ function _fetch_rss_page (feed, page, items, all_items, max_items, oldest_date)
         // number of items, cease crawling
         const done_crawling = (limited_items.length < feed_json.items.length);
 
-        if ((is_paginated || is_array) && !done_crawling) {
-            const new_all_items = new Set(function* () { yield* all_items; yield* new_items.map(item => item.link); }());
+        const new_all_items = new Set(function* () { yield* all_items; yield* new_items.map(item => item.link); }());
```

Comment:
I would move this down closer to where it's used, since it's not used in every code branch here.